### PR TITLE
[SW-2270] Deprecate GridSearch Parameters which Are Exposed also on Algorithms

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala [hunk 1]
```diff
@@ -102,7 +102,50 @@ trait H2OGridSearchParams extends H2OCommonSupervisedParams {
 
   def getSeed(): Long = $(seed)
 
-  def getNfolds(): Int = $(nfolds)
+  @DeprecatedMethod("the 'getPredictionCol' method of a given algorithm", "3.32")
+  def getPredictionCol(): String = getAlgo().getPredictionCol()
```
Comment (jakubhava) [inline ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala:106]: What if the user does:

`grid.setLabelCol("X).setAlgo(new H2OGBM())`. Will label col value be preserved?

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala [hunk 2]
```diff
@@ -77,10 +77,10 @@ class H2OGridSearch(override val uid: String)
       valid: Option[H2OFrame]): Map[String, Any] = {
     algo.getH2OAlgorithmParams() ++
       Map(
-        "nfolds" -> getNfolds(),
-        "fold_column" -> getFoldCol(),
-        "response_column" -> getLabelCol(),
-        "weights_column" -> getWeightCol(),
+        "nfolds" -> getAlgo().getNfolds(),
```
Comment (jakubhava) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala:80]: aren't the params already available in `algo.getH2OAlgorithmParams()` ?

File: py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py [hunk 3]
```diff
@@ -15,12 +15,13 @@
...
 
-from ai.h2o.sparkling.ml.params.H2OCommonSupervisedParams import H2OCommonSupervisedParams
 from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
 from pyspark.ml.param import *
 
+import warnings
 
-class H2OGridSearchParams(H2OCommonSupervisedParams):
+
+class H2OGridSearchParams:
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py:24]: when i running tests locally, it seems it wasn't necessary. i will added back to be safe, thank you!

File: py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py [hunk 4]
```diff
@@ -143,11 +150,94 @@ def getSelectBestModelBy(self):
     def getParallelism(self):
         return self.getOrDefault(self.parallelism)
 
+    def getSeed(self):
+        return self.getOrDefault(self.seed)
+
+    def getLabelCol(self):
+        warnings.warn("The method will be removed in the version 3.32. "
+                      "Use 'getLabelCol' method of a give algorithm instead.", DeprecationWarning)
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py:158]: Thanks!

File: ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala:106]: the `propagateValuesToAlgorithm` dictionary serves for that purpose. Unfortunately, I wasn't able to test it becouse of https://0xdata.atlassian.net/browse/SW-2276

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala (hunk 2 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala:80]: yes, you are right, thanks!

File: ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala:106]: I'm probably missing something -> I can see `propagateValuesToAlgorithm` in Python implementation, but missing it in Scala

Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala:108]: @jakubhava I've tried to address this comment, please can you take a look?

Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala:108]: Thank you, looks good

**Pull request merged**